src/main/java/org/asciidoclet/asciidoclet/AttributesLoader.java

```diff
@@ -65,15 +66,15 @@
     }
 
     private Map<String, Object> parseCmdLineAttributes(List<String> attributeArgs) {
-        return new Attributes(attributeArgs.toArray(new String[attributeArgs.size()])).map();
+        return new Attributes(attributeArgs.toArray( new String[0] )).map();
```

Comment:
This has me a little confused here.

Reply:
This pattern moves the returned array allocation into the `toArray` method, because the zero-length array most likely can't be used as it is too short. When array allocation _and_ population happen together in the same method, the JIT is more likely to infer that the array zeroing isn't necessary. This way, the array is likely only written to once, instead of twice. Recent versions of Intellij IDEA have inspections for pointing this out.

Reply:
Ah, that's cool. I did not know about that short cut!